src/tigerbeetle/repl.zig

```diff
@@ -0,0 +1,813 @@
+                }
+            } else unreachable;
+        };
+
+        const stderr = std.io.getStdErr().writer();
+        try stderr.print("Fail near line {}, column {}:\n\n{s}\n", .{
+            target.position_line,
+            target.position_column,
+            target.line,
+        });
+        var column = target.position_column;
+        while (column > 0) {
+            try stderr.print(" ", .{});
+            column -= 1;
+        }
+        try stderr.print("^ Near here.\n\n", .{});
+        try stderr.print(fmt, args);
+    }
+
+    fn eat_whitespace(parse: *Parse) void {
+        while (parse.offset < parse.input.len and
+            std.ascii.isSpace(parse.input[parse.offset]))
+        {
+            parse.offset += 1;
+        }
+    }
+
+    fn parse_identifier(parse: *Parse) []const u8 {
+        parse.eat_whitespace();
+        const after_whitespace = parse.offset;
+
+        while (parse.offset < parse.input.len and
+            (std.ascii.isAlpha(parse.input[parse.offset]) or
+            parse.input[parse.offset] == '_'))
+        {
+            parse.offset += 1;
+        }
+
+        return parse.input[after_whitespace..parse.offset];
+    }
+
+    fn parse_syntax_char(parse: *Parse, syntax_char: u8) !void {
+        parse.eat_whitespace();
+
+        if (parse.offset < parse.input.len and
+            parse.input[parse.offset] == syntax_char)
+        {
+            parse.offset += 1;
+            return;
+        }
+
+        return error.NoSyntaxMatch;
+    }
+
+    fn parse_value(
+        parse: *Parse,
+    ) []const u8 {
+        parse.eat_whitespace();
+        const after_whitespace = parse.offset;
+
+        while (parse.offset < parse.input.len) {
+            const c = parse.input[parse.offset];
+            if (!(std.ascii.isAlNum(c) or c == '_' or c == '|')) {
+                // Allows flag fields to have whitespace before a '|'.
+                var copy = Parse{ .input = parse.input, .offset = parse.offset };
+                copy.eat_whitespace();
+                if (copy.offset < parse.input.len and parse.input[copy.offset] == '|') {
+                    parse.offset = copy.offset;
+                    continue;
+                }
+
+                // Allow flag fields to have whitespace after a '|'.
+                if (copy.offset < parse.input.len and
+                    parse.offset > 0 and
+                    parse.input[parse.offset - 1] == '|')
+                {
+                    parse.offset = copy.offset;
+                    continue;
+                }
+
+                break;
+            }
+
+            parse.offset += 1;
+        }
+
+        return parse.input[after_whitespace..parse.offset];
+    }
+
+    fn match_arg(
+        out: *ObjectSyntaxTree,
+        key: []const u8,
+        value: []const u8,
+    ) !void {
+        inline for (@typeInfo(ObjectSyntaxTree).Union.fields) |enum_field| {
+            if (std.mem.eql(u8, @tagName(out.*), enum_field.name)) {
```

Comment:
I can't actually do this otherwise I get:

```
/Users/phil/tb/tigerbeetle/src/tigerbeetle/repl.zig:170:17: error: comptime control flow inside runtime block
                continue;
                ^~~~~~~~
```

Or maybe I'm trying to do it wrong.